diff --git nova/scheduler/host_manager.py nova/scheduler/host_manager.py
index 0c64d7a..d5b8aeb 100644
--- nova/scheduler/host_manager.py
+++ nova/scheduler/host_manager.py
@@ -140,7 +140,8 @@ class HostState(object):
 
     def update_from_compute_node(self, compute):
         """Update information about a host from its compute_node info."""
-        if self.updated and self.updated > compute['updated_at']:
+        if (self.updated and compute['updated_at']
+            and self.updated > compute['updated_at']):
             return
         all_ram_mb = compute['memory_mb']
 
